test/library/packages/LinearAlgebra/performance/me/me-perf.chpl

```diff
@@ -0,0 +1,295 @@
+    }
+    writeln("");
+  }
+
+
+
+  {
+    // Input Matrices of Format
+    // [1.0 10^n 0.0]
+    // [0.0 -1.0 0.0]
+    // [0.0  0.0 1.0]
+
+    var b = [1.0, 10.0, 100.0, 1000.0, 10000.0, 100000.0, 1000000.0, 10000000.0, 100000000.0];
+
+    for i in 0..8 do {
+      var X: [Space] eltType;
+      setDiag(X, 0, 1);
+      X(1,1) = -1.0;
+      X(0,1) = b[i];
+
+      var M: [X.domain] eltType;
+
+      for 1..iters {
+        t.start();
+        M = expm(X);
+        t.stop();
+      }
+
+      if !correctness then
+        writeln('LinearAlgebra.expm_10pow' + (i: string) + ": " + ((t.elapsed()/iters): string));
+      else
+        writeln(M);
+
+      t.clear();
+    }
+    writeln("");
+  }
+
+
+  {
+    // Input Matrix of Format
+    // [ 0.0  1.0  0.0]
+    // [-1.0  0.0  1.0]
+    // [ 0.0 -1.0  0.0]
+
+    var X: [Space] eltType;
+    setDiag(X, 0, 0.0);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] eltType;
+
+    for 1..iters {
+      t.start();
+      M = expm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.expm_gradient: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
+
+
+  {
+    // Input Matrix of Format
+    // [-2.0  1.0  0.0]
+    // [-1.0 -2.0  1.0]
+    // [ 0.0 -1.0 -2.0]
+
+    var X: [Space] eltType;
+    setDiag(X, 0, -2.0);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] eltType;
+
+    for 1..iters {
+      t.start();
+      M = expm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.expm_edge: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
+
+  {
+    // Testing eltType=Complex: Input Matrix of Format
+    // [ 1.0 + 1i 1.0       0.0     ]
+    // [-1.0      1.0 + 1i  1.0     ]
+    // [ 0.0     -1.0       1.0 + 1i]
+
+    var X: [Space] complex;
+    setDiag(X, 0, 1.0 + 1i);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] complex;
+
+    for 1..iters {
+      t.start();
+      M = expm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.expm_complex: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
+}
+
+proc test_sinm() {
+  {
+    // Testing eltType=Complex: Input Matrix of Format
+    // [ 1.0 + 1i 1.0       0.0     ]
+    // [-1.0      1.0 + 1i  1.0     ]
+    // [ 0.0     -1.0       1.0 + 1i]
+
+    const Space = {0..<m, 0..<m};
+    var t: Timer;
+
+    var X: [Space] complex;
+    setDiag(X, 0, 1.0 + 1i);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] complex;
+
+    for 1..iters {
+      t.start();
+      M = sinm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.sinm: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
```

Comment:
Can you remove these inner blocks in each test?

Reply:
Hi Engin, I used the inner blocks in order to reuse the variable names like the array "b". (which mostly had different sizes for each case). If you think that makes a difference in the performance then I'll remove it.

Reply:
It is not about performance.

But I don't follow, this block is the same as the function body? Ah, I see. The functions starting with this one doesn't need such a block.